crates/fuzzing/src/generators/module.rs

```diff
@@ -16,27 +16,42 @@ impl<'a> Arbitrary<'a> for ModuleConfig {
     fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<ModuleConfig> {
         let mut config = wasm_smith::Config::arbitrary(u)?;
 
+        // This list is intended to be the definintive source of truth for
+        // what's at least possible to fuzz within Wasmtime. This is a
+        // combination of features in `wasm-smith` where some proposals are
+        // on-by-default (as determined by fuzz input) and others are
+        // off-by-default (as they aren't stage4+). Wasmtime will default-fuzz
+        // proposals that a pre-stage-4 to test our own implementation. Wasmtime
+        // might also unconditionally disable proposals that it doesn't
+        // implement yet which are stage4+. This is intended to be an exhaustive
+        // list of all the wasm proposals that `wasm-smith` supports and the
+        // fuzzing status within Wasmtime too.
+        let _ = config.multi_value_enabled;
+        let _ = config.saturating_float_to_int_enabled;
+        let _ = config.sign_extension_ops_enabled;
+        let _ = config.bulk_memory_enabled;
+        let _ = config.reference_types_enabled;
+        let _ = config.simd_enabled;
+        let _ = config.relaxed_simd_enabled;
+        let _ = config.tail_call_enabled;
+        config.exceptions_enabled = false;
+        config.gc_enabled = false;
+        config.wide_arithmetic_enabled = u.arbitrary()?;
+        config.memory64_enabled = u.ratio(1, 20)?;
+        // Allow the threads proposal if memory64 is not already enabled. FIXME:
+        // to allow threads and memory64 to coexist, see
+        // https://github.com/bytecodealliance/wasmtime/issues/4267.
+        config.threads_enabled = !config.memory64_enabled && u.ratio(1, 20)?;
+        // FIXME: this may be safe to enable
+        config.custom_page_sizes_enabled = false;
```

Comment:
cc @fitzgen on this point, I left this as it was before but you might be interested in changing this to `u.arbitrary()?` to have it on-by-default in fuzzing. (will probably require disabling it in other differential fuzzers and maybe winch too)